doc/api/url.md

```diff
@@ -250,7 +250,308 @@ properties of URL objects:
+myURL.search = 'abc=xyz';
+console.log(myURL.href);
+  // Prints http://example.org/abc?abc=xyz
+```
+
+#### url.searchParams
+
+Gets a `URLSearchParams` object representing the querystring parameters of the
+URL.
+
+#### url.username
+
+Gets and sets the username portion of the URL.
+
+```js
+const myURL = new URL('http://abc:xyz@example.com');
+console.log(myURL.username);
+  // Prints abc
+
+myURL.username = '123';
+console.log(myURL.href);
+  // Prints http://123:xyz@example.com
+```
+
+#### url.toString([unicode])
+
+* `unicode` {Boolean} `true` to serialize Unicode characters in the URL
+  hostname directly. Passing `false` causes the punycode algorithm to be
+  used.
+
+The `toString()` method on the `URL` object returns the serialized URL.
+
+### Class: URLSearchParams
+
+The `URLSearchParams` object provides read and write access to the querystring
+of a `URL`.
+
+```js
+const myURL = new URL('http://example.org/?abc=123');
+console.log(myURL.searchParams.get('abc'));
+  // Prints 123
+
+myURL.searchParams.append('abc', 'xyz');
+console.log(myURL.href);
+  // Prints http://example.org/?abc=123&abc=xyz
+
+myURL.searchParams.delete('abc');
+myURL.searchParams.set('a', 'b');
+console.log(myURL.href);
+  // Prints http://example.org/?a=b
+```
+
+#### Constructor: new URLSearchParams([init])
+
+* `init` {String} The URL querystring
+
+#### urlSearchParams.append(name, value)
+
+* `name` {String}
+* `value` {String}
+* Returns `undefined`
+
+Append a new name-value pair to the query string.
+
+#### urlSearchParams.delete(name)
+
+* `name` {String}
+* Returns `undefined`
+
+Remove all name-value pairs whose name is `name`.
+
+#### urlSearchParams.get(name)
+
+* `name` {String}
+* Returns a string or `undefined`
+
+Returns the value of the first name-value pair whose name is `name`.
+
+#### urlSearchParams.getAll(name)
+
+* `name` {String}
+* Returns a JavaScript Array
+
+Returns the values of all name-value pairs whose name is `name`.
+
+#### urlSearchParams.has(name)
+
+* `name` {String}
+* Returns `true` or `false`
+
+Returns `true` if there is at least one name-value pair whose name is `name`.
+
+#### urlSearchParams.set(name, value)
+
+* `name` {String}
+* `value` {String}
+* Returns `undefined`
+
+Remove any existing name-value pairs whose name is `name` and append a new
+name-value pair.
+
+#### urlSearchParams[Symbol.iterator]()
+
+Retuns an ES6 Iterator over each of the name-value pairs in the query string.
+Each item of the iterator is a JavaScript Array. The first item of the Array
+is the `name`, the second item of the Array is the `value`.
+
+### require('url').originFor(url)
+
+* `url` {String | URL}
+
+Returns an object representing the origin of the given URL. The origin object
+is considered to be opaque. That is, while there are properties and methods
+exported by the object, they are not considered to be part of the "public"
+API of the object.
+
+*Note*: The origin object and the `require('url').originFor()` API are not part
```

Comment:
No, these are added as part of the new experimental implementation. They are just not part of the standard